### PR TITLE
refactor: sync with axon new start commands

### PR DESCRIPTION
**Pull request by wenyuanhust**
## Description

1. sync with latest version of axon (10.23), the start process has changed
2. upgrade crate axon-tools
    - sync with latest version of axon (10.23), data structures like Block, Proof, Proposal has changed
    - verfiy_proof api used for block verfication updated
    - verify_trie_proof api used for tx verfication enabled and updated

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Forcerelay) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: .github/workflows/ibc-test.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     timeout-minutes: 60
     env:
       SRC_DIR: ${{ github.workspace }}/ibc-test-src
-      AXON_COMMIT: d03d2bb7cb3dcdc03319c3a74beeee6715e7f448
+      AXON_COMMIT: 7ae0a826647d07e0b25b4adb440a8c97231d6a04
```
Comment (Flouse) [inline .github/workflows/ibc-test.yaml]: Currently, the Axon version should contain this commit to [change the create contract limit config](https://github.com/axonweb3/axon/commit/56b1a218ea5ed0b84aae5e74422b4f77bcfc8b6e).

Would you try this Axon branch?
-> https://github.com/axonweb3/axon/commits/0.2.0-dev.1

**Review by Flouse — CHANGES_REQUESTED** [review]

File: .github/workflows/ibc-test.yaml (hunk 1 above)
Comment (Flouse) [reply .github/workflows/ibc-test.yaml]: Related issue:
- https://github.com/synapseweb3/ibc-solidity-contract/issues/9

**Review by Flouse — DISMISSED** [review]
Please try to use the latest release of Axon.
@wenyuanhust

**Comment by ashuralyk** [comment]
when we upgrade `axon-tools` dependency, the matched axon binary should be initialed with its new start command, so if we plan to test it in CI after the integration of `axon-tools`, this PR is needed to merge

**Review by ashuralyk — APPROVED** [review]

**Review by blckngm — APPROVED** [review]

**Pull request merged**